Add a draft of Rust Project contributor survey 2025

Kobzol opened the pull request:
This is a draft of a survey targeted at Rust Project members, to find out how much of their contribution time is being funded, vs how much of it is volunteer based (https://github.com/rust-lang/leadership-council/issues/222).

Some notes:
- Currently, the survey is only targered at Rust Project *members* (so members of some team/wg/etc., pretty much anyone who was invited to the Rust AllHands), not all Rust Project contributions. The idea is that those are the most people who would appreciate funding the most, as they usually contribute a lot, do a lot of maintenance, non-technical work, etc. But we could also make this open to any Rust Project contributor.
- Currently I describe at the beginning that he survey is only targered to people who contribute to the Rust Project. We don't want other people filling the survey, ofc. We could either rely on people not abusing this, or we could share the link e.g. in a private Zulip channel where there are only members of `t-all` or make the survey password-protected (but then we need to share the password through a similar channel anyway).
- I did not include questions about whether the person is employed or not. I don't think that we need this information? And that the thing that we really care about is how much of their contributions are funded. If someone does Rust work as a volunteer, do we care whether they are a student vs if they are employed?
- I included a set of three personal questions at the end, which could allow us to find out more interesting information about how much money do people receive, which companies are funding them, and then even separate this information across various teams, as we'd also have the user's GH username. If we include these questions, we'd most likely need to run the survey in cooperation with the Rust Foundation. I was told by Paul Lenz that Lori might be able to help us with that.

davidtwco reviewed (commented):
This is great, thanks for working on this @Kobzol, some thoughts:

- I think we should ask if someone has previously been paid to contribute to Rust.

    - It would be useful to capture whether there are project members who have previously been funded to contribute and no longer are - repeating this survey over time will capture this, but for the first run, we might ask. We might also ask why - did grants run out? contracts didn't get renewed? was someone laid off? We *could* ask where these previous funding sources came from (going forward, maybe since the last of these surveys that the contributor completed).
    - We could also learn whether there are contributors that have had multiple sources of funding over time while there are contributors who don't have any opportunities (that might be expected, one would hope our most very experienced language team members, for example, to always have opportunities like this, but it is interesting nevertheless).

- We should also ask what team people are in (and not just rely on the username to work this out, in case someone would prefer not just to share that).

    - There is very likely to be an uneven distribution of contributor support, and that might be useful to understand.

- We could ask whether contributions are directed by their source of funding.

    - e.g. at Arm, we focus on Arm support in the toolchain, but other contributors direct their own work, or work on things useful to their employer's users of Rust but aren't employer-specific as such.

- You've said you're not interested in capturing this in the PR description, but I think it would be useful to know whether someone contributes full-time/part-time/etc.
    
    - I think that's subtly different from what % of contributions are funded or the source of funding (employment/contracts/etc). Someone might have 100% of their contributions funded by a contract, but that could be only four hours each week and they still need to work another job and do Rust in the evenings and weekends - that still impacts the sustainability of their contributions and we should understand that.

- If someone has funding, we should ask how long they have been funded from their current sources and how long that funding lasts (where applicable).

    - It would be interesting to know if everyone is on 1yr contracts that might not be renewed. Similarly, if lots of people have been in their positions for many years, that implies a stability that is useful to know.

- We could ask whether the non-Rust contribution part of their role is related to Rust? Or whether their source of funding is a user of Rust?

    - Contributors often also support internal use of Rust with some of their time, this is leveraging the experience of the contributor with the project and language, and knowing that is a dynamic at play might be useful?
    - Similarly, knowing whether most of contributor funding comes from users of the language supporting it, compared to grants/funds that are from foundations/organisations independent of language, might be a proxy for the sustainability of that source of funding (in principle, if you use Rust then you're probably likely to continue supporting it compared to if you don't care about Rust as such and were funding a specific project).

- For those who don't need funding to contribute, we could try and determine how they make that sustainable?

    - e.g. someone is an academic or student and so can do it on the side easily enough, or something like that.

- If we're going to ask about the dollar amounts of funding received, then asking for a location might make sense - as that will help contextualise that amount.

    - e.g. contributors in Europe are likely paid less than their North American counterparts and that could explain some variation in the results to that question

- Is it worth trying to reach people funded to work on Rust but who aren't project members?

    - I agree it makes sense to limit this to project members, not irregular contributors, to  get a better signal-to-noise ratio. Ultimately, not everyone will be able to get funding to work on Rust, but we want as many of the core contributors to the toolchain to have funding, where the impact should be greatest and that funding is most deserved, so using project members as a proxy for that makes sense. However, I'm aware of people who are paid to work on Rust full-time but aren't project members, which might be an interesting niche to understand more? Where is that funding coming from? Could it be distributed more effectively? etc.

Kobzol commented:
> I think we should ask if someone has previously been paid to contribute to Rust.

Good idea, added.

> We should also ask what team(s) people are in (and not just rely on the username to work this out, in case someone would prefer not just to share that).

Fair enough, added.

> We could ask whether contributions are directed by their source of funding.

Also a good idea, added. Although the specific wording of answers might need some more work.

> You've said you're not interested in capturing this in the PR description, but I think it would be useful to know whether someone contributes full-time/part-time/etc.

Hmm. My assumption was that if you're able to get funding for your Rust contributions, it would naturally diminish the time you have to spend working on another job. I worry a bit that this survey is becoming a bit larger than I originally hoped, but if you could suggest a specific question/answer set/framing to put into the updated version of the survey, I can add it.

> If someone has funding, we should ask how long they have been funded from their current sources and how long that funding lasts (where applicable).

Added.

> We could ask whether the non-Rust contribution part of their role is related to Rust? Or whether their source of funding is a user of Rust?

I think that this question is starting to hit diminishing returns of what we want to figure out in this survey, and I don't know how to frame it. I'd rather let people describe the specifics in the final open answer.

> For those who don't need funding to contribute, we could try and determine how they make that sustainable?

Added.

> If we're going to ask about the dollar amounts of funding received, then asking for a location might make sense - as that will help contextualise that amount.

Added.

> However, I'm aware of people who are paid to work on Rust full-time but aren't project members, which might be an interesting niche to understand more?

Maybe, but I'm not sure if this survey is the right avenue for that. IMO we should focus on the use-case on supporting Project members that work on Rust and have not enough funding, rather than examining companies that pay people to work on Rust from outside the Project.

> We could also ask, esp. if someone isn't willing to share the name of their employer/source of funding, if that source of funding is a member of the Foundation or not?

Would we make that information public? I think that the only reason not to share the information is if we made it public or mappable to individual people. If we ensure that the information will not be shared with anyone outside the Council/RF, then I hope people wouldn't hesitate to share the company names. More generally, I worry about the survey becoming a bit large, as noted above.

Kobzol commented:
Currently, we have 16 questions (out of which 4 are personal/heavily optional). For context, the compiler performance survey had 28, the annual survey has 50, the most recent micro survey had ~13. I think that we can make this survey a bit longer, as I expect/hope that Rust Project contributions will spend enough time on this, but I also don't want to make the survey too complicated :sweat_smile:

Kobzol commented:
I was just thinking today that we need to have this question :+1: I'll add it.

davidtwco commented on surveys/2025/contributor-survey/questions.md:
```diff
@@ -0,0 +1,181 @@
+
+### How much of your time spent on Rust Project contributions is funded?
+
+Estimate how much of the time that you spend on contributing to the Rust Project is funded. For example, if you spend 20 hours per week contributing, and a company pays you for 10 hours of contributions per week, then approx. 50% of your contribution time is funded.
```
This is probably one of the more complex questions and might require a bit more guidance or examples of how to answer - I'm trying to think about how I'd answer it: I'm salaried for 38hr/wk, and all I work on are Rust contributions, but also have internal meetings and so on. Do I say that I spend 38hr/wk is contributing, do I remove ~5-6hrs/wk for the internal bits and say I'm contributing that much? It might be worth having some examples of common scenarios and how we'd like people to answer to encourage uniformity.

Kobzol replied:
Did you mean to put this to the "How much time do you spend contributing to the Rust Project monthly?" question?

For this question, we only want to know the ratio. So if you work 33hr/wk on Rust, then 100% of your time is funded, and you should answer 100.

davidtwco replied:
Oops, yeah, I intended to put this on "How much time do you spend contributing to the Rust Project monthly?"

davidtwco replied:
I meant examples in the sense of "common contributor funding arrangements" (salaried/contractor/etc) rather than "types of contribution" - e.g. if you're employed at a company and spend a few hours each week on internal meetings or 1-1s or whatever, you do/don't need to try and work out the actual number of hours contributing. Stuff like that, or similar things for other arrangements - just so that there's a little bit of standardisation as far as how people decide to tell us those details.

Kobzol replied:
Tried to clarify the meetings, let me know if you have suggestions for further edits!

oli-obk commented on surveys/2025/contributor-survey/questions.md:
```diff
@@ -0,0 +1,186 @@
+Type: select one (optional)
+
+- Yes, my source of funding tells me what contributions I should make
+- Somewhat, my work is useful to my source of my funding, but I mostly decide what to contribute
```
I think this option can just be part of the "other"

apiraino commented on surveys/2025/contributor-survey/questions.md:
```diff
@@ -0,0 +1,186 @@
+Type: select all that apply (optional)
+
+- I make enough money on the side and have enough free time for contributions
+- I am a student and have enough free time for contributions
+- I am an academic and have enough free time for contributions
```
> I think it might be interesting to figure out how are contributors able to spend a lot of time working on Rust without getting a financial reward.

I could have inherited a billion dollar from my dead aunt or I did a rich exit from a startup. I have FU money so I can contribute to Rust forever.

I just don't think it's interesting, that's all :slightly_smiling_face:

Kobzol replied:
I think a more common situation might be something like a student that might not even think that they *deserve* to be paid for working on Rust, which is of course a situation that we'd like to avoid.

workingjubilee commented:
> For those who don't need funding to contribute, we could try and determine how they make that sustainable?

I mean, the answer could be "It's not".

Kobzol commented:
That's a good point, I'll add it.

By the way, today I was reminded of another Contributor survey that was designed in 2021 (https://github.com/rust-lang/surveys/blob/main/surveys/2021-contributor-survey/design.md, https://github.com/rust-lang/surveys/blob/main/surveys/2021-contributor-survey/questions.md), which was never actually published :( It was much broader than this survey. But if someone finds some interesting questions there that we could pluck into this one, let me know.

I think that it might be interesting to ask what are the specific actions that people work on (reviews, implementation, triaging, infra, etc.). That might show some patterns like "people doing triaging are essentially unfunded" or something like that.

apiraino commented on surveys/2025/contributor-survey/questions.md:
```diff
@@ -0,0 +1,230 @@
+- That sounds like a good idea, and I would potentially like to be on that list
+- That sounds like a good idea, but I do not want to be on such a list
+- I'm neutral/I don't know
+- That sounds like a bad idea
```
maybe?

```suggestion
- That sounds like a bad idea (please explain why)
```

Kobzol replied:
Good point, I was also thinking about that, but I'm not sure how to combine "select one answer" + allow an open answer, hmm.

Kobzol replied:
I tried this as a compromise: https://github.com/rust-lang/surveys/pull/350/commits/fac52ae1bea59d1d3342f4c63218529bdc6256d9

apiraino replied:
The wording is fine but if I read correctly there is no text field where people can write ~~things~~ the "why", right?

Kobzol commented:
I wanted to ask the council about feedback (gonna do that hopefully at the next meeting, starting in exactly one minute :) ). We didn't get to it the last time.

Kobzol commented:
Based on the discussions around maintainer funding, I added a question about preferred funding arrangement (employment/contracting etc.).
